### PR TITLE
Introduce gosec for security checks

### PR DESCRIPTION
**Pull request by JAORMX**
gosec does static code analysis and checks for common security issues in
golang codebases.

This PR introduces the tool, and sets it up as a test target in the Makefile.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,6 +30,11 @@ vet: ## Apply go vet to all go files
 unit: # Run unit test
 	$(DOCKER_CMD) go test -race -cover ./cmd/... ./pkg/...
 
+.PHONY: sec
+sec: # Run security static analysis
+	go get -u github.com/securego/gosec/cmd/gosec
+	gosec -severity medium -confidence medium -exclude G304 -quiet ./...
```
Comment (ingvagabund) [inline Makefile]: Would it be possible to run the command from within a container?

Comment (JAORMX) [reply Makefile]: It is possible. That's actually what was proposed for the installer: https://github.com/openshift/installer/pull/2126/files

**Comment by JAORMX** [comment]
Any more feedback on this PR?

File: Makefile (hunk 1 above)
Comment (ingvagabund) [reply Makefile]: Can you update the PR so `gosec` is run inside a container?

Comment (JAORMX) [reply Makefile]: Done! thanks for the feedback

**Pull request merged**